src/actions/tests/ast.js

```diff
@@ -0,0 +1,33 @@
+import {
+  createStore,
+  selectors,
+  actions,
+  makeSource
+} from "../../utils/test-head";
+
+const { getSource, getSourceText, getSymbols } = selectors;
+
+const threadClient = {
+  sourceContents: function(sourceId) {
+    return new Promise((resolve, reject) =>
+      resolve({
+        source: sourceTexts[sourceId],
+        contentType: "text/javascript"
+      })
+    );
+  }
+};
+
+const sourceTexts = {
+  "base.js": "function base(boo) {}"
+};
+
+describe("ast", () => {
+  it("should be able to set symbols for a source", async () => {
+    const { dispatch, getState } = createStore(threadClient);
+    await dispatch(actions.newSource(makeSource("base.js")));
+    await dispatch(actions.loadSourceText({ id: "base.js" }));
+    const baseSymbols = getSymbols(getState(), "base.js");
+    expect(baseSymbols).toMatchSnapshot();
```

Comment:
the snapshot is great because we don't care what the symbols are. jus that they're sensible